tests/foreman/cli/test_remoteexecution.py

```diff
@@ -1051,3 +1050,98 @@ def test_positive_run_job_on_host_registered_to_pull_provider(
+        sat = module_provisioning_sat_pull_provider.sat
+
+        host = sat.api.Host(
+            hostgroup=module_provisioning_rhel_content.hostgroup,
+            organization=module_org_with_manifest,
+            location=module_location,
+            content_facet_attributes={
+                'content_view_id': module_default_org_view.id,
+                'lifecycle_environment_id': module_lce_library.id,
+            },
+            name=gen_string('alpha').lower(),
+            mac=host_mac_addr,
+            operatingsystem=module_provisioning_rhel_content.os,
+            subnet=module_provisioning_sat_pull_provider.subnet,
+            host_parameters_attributes=[
+                {
+                    'name': 'remote_execution_connect_by_ip',
+                    'value': 'true',
+                    'parameter_type': 'boolean',
+                },
+                {
+                    'name': 'enable-remote-execution-pull',
+                    'value': 'true',
+                    'parameter_type': 'boolean',
+                },
+            ],
+            build=True,  # put the host in build mode
+        ).create(create_missing=False)
+
+        # Clean up the host to free IP leases on Satellite.
+        # broker should do that as a part of the teardown, putting here just to make sure.
+        request.addfinalizer(host.delete)
+
+        # Start the VM, do not ensure that we can connect to SSHD
+        provisioning_host.power_control(ensure=False)
+
+        # Host should do call back to the Satellite reporting
+        # the result of the installation. Wait until Satellite reports that the host is installed.
+        wait_for(
+            lambda: host.read().build_status_label != 'Pending installation',
+            timeout=1500,
+            delay=10,
+        )
+        host = host.read()
+        assert host.build_status_label == 'Installed'
+
+        # Change the hostname of the host as we know it already.
+        # In the current infra environment we do not support
+        # addressing hosts using FQDNs, falling back to IP.
+        provisioning_host.hostname = host.ip
+        # Host is not blank anymore
+        provisioning_host.blank = False
+
+        # check ygg client
+        result = provisioning_host.execute('yggdrasil status')
+        assert result.status == 0, f'Failed to start yggdrasil on client: {result.stderr}'
+        result = provisioning_host.execute('systemctl status yggdrasild')
+        assert result.status == 0, f'Failed to start yggdrasil on client: {result.stderr}'
+
+        # run script template
+        invocation_command = make_job_invocation(
+            {
+                'job-template': 'Run Command - Script Default',
+                'inputs': 'command=ls',
+                'search-query': f"name ~ {provisioning_host.hostname}",
+            }
+        )
+        assert_job_invocation_result(invocation_command['id'], host.hostname)
```

Comment:
Just a thought... Does it make sense to add an explicit check/assert that the job invocation used pull provider instead of SSH?
If so, how to check it? On server or on client?

Reply:
I agree that such an assertion makes sense.
I'd probably check some logs on the client side but i'm not familiar with the implementation of the feature